packages/components/addon/modifiers/hds-tooltip.js

```diff
@@ -0,0 +1,162 @@
+    name: 'hideOnEsc',
+    defaultValue: true,
+    fn({ hide }) {
+      function onKeyDown(event) {
+        if (event.keyCode === 27) {
+          hide();
+        }
+      }
+
+      return {
+        onShow() {
+          document.addEventListener('keydown', onKeyDown);
+        },
+        onHide() {
+          document.removeEventListener('keydown', onKeyDown);
+        },
+      };
+    },
+  };
+
+  modify(element, positional, named) {
+    assert('Tooltip must have an element', element);
+
+    if (!this.didSetup) {
+      this.#setup(element, positional, named);
+      this.didSetup = true;
+    }
+
+    this.#update(element, positional, named);
+  }
+
+  #setup(element, positional, named) {
+    const tooltipProps = this.#getTooltipProps(element, positional, named);
+    this.tooltip = tippy(element, tooltipProps);
+  }
+
+  #update(element, positional, named) {
+    const tooltipProps = this.#getTooltipProps(element, positional, named);
+    this.tooltip.setProps(tooltipProps);
+  }
+
+  #getTooltipProps(element, positional, named) {
+    const { options = {} } = named;
+    let [content] = positional;
+
+    let $anchor = element;
+
+    // Make it easy to specify the modified element as the actual tooltip.
+    if (typeof options.triggerTarget === 'string') {
+      const $el = $anchor;
+      switch (options.triggerTarget) {
+        case 'parentNode':
+          $anchor = $anchor.parentNode;
+          break;
+        default:
+          $anchor = $anchor.querySelectorAll(options.triggerTarget);
+      }
+      content = $anchor.cloneNode(true);
+      $el.remove();
+      options.triggerTarget = undefined;
+    }
+
+    // The {{hds-tooltip}} will just use the HTML content.
+    if (typeof content === 'undefined') {
+      content = $anchor.innerHTML;
+      $anchor.innerHTML = '';
+    }
+
+    if (options.trigger === 'manual') {
+      // If we are manually triggering, a out delay means only show for the
+      // amount of time specified by the delay.
+      const delay = options.delay || [];
+
+      if (typeof delay[1] !== 'undefined') {
+        options.onShown = (tooltip) => {
+          clearInterval(this.interval);
+          this.interval = setTimeout(() => {
+            tooltip.hide();
+          }, delay[1]);
+        };
+      }
+    }
+
+    const $trigger = $anchor;
+
+    if (!$trigger.hasAttribute('tabindex')) {
+      this.needsTabIndex = true;
+      $trigger.setAttribute('tabindex', '0');
+    }
+
+    return {
+      theme: 'hds',
+      triggerTarget: $trigger,
+      // trigger: 'click', // TEMP for testing design
+      arrow: `
+        <svg width="16" height="7" viewBox="0 0 16 7" xmlns="http://www.w3.org/2000/svg">
+          <path d="M0 7H16L9.11989 0.444571C8.49776 -0.148191 7.50224 -0.148191 6.88011 0.444572L0 7Z" />
+        </svg>`,
+      // keeps tooltip itself open on hover:
+      interactive: true,
+      // fix accessibility features that get messed up with setting interactive: true
```

Comment:
They (the Tippy.js people) are assuming that this means there will be interactive elements inside the tooltip so they change the accessibility roles to reflect that. In our case, we are only making use of this feature so that the tooltip remains open when you hover over it and we don't intend for there to be interactive content inside it so we have to restore the related roles...

Reply:
makes sense – thanks for expanding on this